### PR TITLE
CheckIn

### PR DESCRIPTION
**Pull request by Pineirin**
Add autofocus to the input form. Invert order of the returned loans list and highlight the location of each book in that list. Visual ui changes.
closes https://github.com/CERNDocumentServer/cds-ils/issues/531
Screenshot from 2021-08-03 13-08-37

### CODE REVIEW

File: src/lib/pages/backoffice/Actions/CheckIn/CheckedInItems/CheckedInItems.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ export default class CheckedInItems extends Component {
 
   renderItems = () => {
     const { items } = this.props;
-    const entries = items.map((item) => {
+
+    const entries = items.reverse().map((item) => {
```
Comment (ntarocco) [inline src/lib/pages/backoffice/Actions/CheckIn/CheckedInItems/CheckedInItems.js:16]: This is probably ok in most of the cases, but it might not be enough.
Here you are reversing the list client side, in place. The issue is that Elasticsearch will only give you the first `x` results. If the item to have as first is at the `x+1` position, the code above will not work because Elasticsearch did not return it.

It would be better to change the Elasticsearch query sort order instead, if possible.

File: src/semantic-ui/site/elements/list.overrides [hunk 2]
```diff
@@ -32,6 +32,10 @@ List Overrides - REACT-INVENIO-APP-ILS
     padding: 0.5em;
   }
 
+  &.metadata-sharp{
```
Comment (ntarocco) [inline src/semantic-ui/site/elements/list.overrides:35]: To be discussed if bold+black is enough. In any case, instead of `sharp`, why not creating a class `.color-black` that can be eventually re-used?

File: src/lib/components/SearchBar/SearchBarILS.js [hunk 3]
```diff
@@ -65,13 +55,6 @@ export class SearchBarILS extends Component {
         size="big"
         placeholder={placeholder}
         className={`${parentClass} ils-searchbar`}
-        ref={(input) => {
```
Comment (kprzerwa) [inline src/lib/components/SearchBar/SearchBarILS.js:68]: are you sure we want to remove the focus from this component which is widely used across the whole application ?

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 4]
```diff
@@ -75,25 +75,29 @@ export class ItemListEntry extends Component {
...
                     truncateLines={1}
                   />
                 </Header>
-                <List>
+                <List className="metadata-sharp">
                   <List.Item>
                     <List.Content>
-                      {item.metadata.internal_location.location.name} -{' '}
-                      {item.metadata.internal_location.name}
+                    <div class="ui label">
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:89]: in this way you are using the pure semantic ui, which is not what we want to do, since we have a library which provides the components for react. Please check react semantic ui library documentation for more details.

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 5]
```diff
@@ -75,25 +75,29 @@ export class ItemListEntry extends Component {
...
+            <Grid.Column computer={5} largeScreen={5}>
               <Item.Meta className="metadata-fields">
                 <Header as="h5">
                   <LiteratureTitle
                     title={item.metadata.document.title}
                     truncateLines={1}
                   />
                 </Header>
-                <List>
+                <List className="metadata-sharp">
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:86]: do we still need this class, since below you use `<strong>` ?

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 6]
```diff
@@ -75,25 +75,29 @@ export class ItemListEntry extends Component {
...
+                      <strong>{item.metadata.internal_location.location.name} -{' '}
+                      {item.metadata.internal_location.name}</strong>
+                    </div>
                     </List.Content>
                   </List.Item>
                   <List.Item>
                     <List.Content>
                       {' '}
-                      shelf <strong>{item.metadata.shelf}</strong>
+                      <div class="ui label">
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:98]: same as above

File: src/semantic-ui/site/elements/list.overrides [hunk 7]
```diff
@@ -32,6 +32,12 @@ List Overrides - REACT-INVENIO-APP-ILS
     padding: 0.5em;
   }
 
+  &.metadata-sharp > * > * > *{
```
Comment (kprzerwa) [inline src/semantic-ui/site/elements/list.overrides:35]: in terms of CSS and semantic UI combination, the parent child selectors usage is quite dangerous - if someone changes the structure, or in the new version of semantic UI we will have more elements in this hierarchy, this stylesheet will not be applied correctly. is there any existing out of the box component you could use instead of the custom CSS? (custom CSS should be used only as a last resort, if there is no other solution)

File: src/lib/components/SearchBar/SearchBarILS.js [hunk 8]
```diff
@@ -1,18 +1,10 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { Input } from 'semantic-ui-react';
-import { screenIsWiderThan } from '@components/utils';
-import { Breakpoints } from '@components/Media';
 
 export class SearchBarILS extends Component {
   state = { currentValue: '' };
 
-  componentDidMount() {
```
Comment (kprzerwa) [inline src/lib/components/SearchBar/SearchBarILS.js:10]: are you sure we want this removed ?

File: src/lib/components/SearchBar/SearchBarILS.js [hunk 9]
```diff
@@ -43,11 +35,9 @@ export class SearchBarILS extends Component {
       onPasteHandler,
       onChangeHandler,
       placeholder,
-      responsiveAutofocus,
       ...rest
     } = this.props;
     const { currentValue } = this.state;
-    const autofocus = screenIsWiderThan(Breakpoints.computer);
```
Comment (kprzerwa) [inline src/lib/components/SearchBar/SearchBarILS.js:50]: and this one ?

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 10]
```diff
@@ -140,20 +150,12 @@ export class ItemListEntry extends Component {
...
-                to={BackOfficeRoutes.documentDetailsFor(
-                  item.metadata.document_pid
-                )}
-              >
-                <DocumentIcon /> Document
-              </Link>
               {withPendingLoans && (
                 <Item.Extra>
                   <Button
+                    primary
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:156]: 👍

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 11]
```diff
@@ -78,22 +78,33 @@ export class ItemListEntry extends Component {
             <Grid.Column computer={6} largeScreen={6}>
               <Item.Meta className="metadata-fields">
                 <Header as="h5">
+                <Link
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:81]: how this influence the Items search view ? did you have any notification of changed snaphots in ItemSearch (in tests)?

File: src/semantic-ui/site/views/item.overrides [hunk 12]
```diff
@@ -131,7 +125,7 @@ Item Overrides - REACT-INVENIO-APP-ILS
       }
 
       &.bkg-yellow {
-        background-color: #fff4bd !important;
+        background-color: ##ffffe6 !important;
```
Comment (kprzerwa) [inline src/semantic-ui/site/views/item.overrides:128]: does it affect any other places in the application ?

Comment (Pineirin) [reply src/semantic-ui/site/views/item.overrides:128]: No

File: src/lib/modules/Items/backoffice/ItemListEntry.js (hunk 11 above)
Comment (Pineirin) [reply src/lib/modules/Items/backoffice/ItemListEntry.js:81]: I didn't, it seems to work fine, but I will double-check.

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 13]
```diff
@@ -167,7 +204,11 @@ export class ItemListEntry extends Component {
             </Grid.Column>
           </Grid>
         </Item.Content>
-        <div className="pid-field">#{item.metadata.pid}</div>
+        {checkIn ? (
+          <div className="pid-field pid-field-top">#{item.metadata.pid}</div>
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:208]: actually it would be enough if you just removed all the classes here, without creating a new one

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 14]
```diff
@@ -78,23 +79,57 @@ export class ItemListEntry extends Component {
             <Grid.Column computer={6} largeScreen={6}>
               <Item.Meta className="metadata-fields">
                 <Header as="h5">
-                  <LiteratureTitle
-                    title={item.metadata.document.title}
-                    truncateLines={1}
-                  />
+                  {checkIn ? (
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:82]: I think this change should be reverted, since it does not add so much in terms of the user experience, and it might be confusing to have this element in different places, depending on the location. I think we should keep things as much harmonised as possible

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 15]
```diff
@@ -78,23 +79,57 @@ export class ItemListEntry extends Component {
...
+                    />
+                  )}
                 </Header>
                 <List>
                   <List.Item>
-                    <List.Content>
-                      {item.metadata.internal_location.location.name} -{' '}
-                      {item.metadata.internal_location.name}
-                    </List.Content>
+                    {checkIn ? (
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:103]: you are on a good track.
having in mind the previous comment, I would change `checkin` prop name to `highlightLocation` and use it only for this purpose. It might come up handy and more flexible if we want to highlight the location in the other places this component is used.

File: src/lib/modules/Items/backoffice/ItemListEntry.js [hunk 16]
```diff
@@ -140,20 +175,22 @@ export class ItemListEntry extends Component {
...
             <Grid.Column computer={2} largeScreen={2} textAlign="right">
-              <Link
-                target="_blank"
-                to={BackOfficeRoutes.documentDetailsFor(
-                  item.metadata.document_pid
-                )}
-              >
-                <DocumentIcon /> Document
-              </Link>
+              {!checkIn && (
```
Comment (kprzerwa) [inline src/lib/modules/Items/backoffice/ItemListEntry.js:178]: I would revert this change, check the first comment for the explanation

File: src/lib/pages/backoffice/Actions/CheckIn/CheckedInItems/CheckedInItems.js [hunk 17]
```diff
@@ -19,6 +19,7 @@ export default class CheckedInItems extends Component {
           key={item.id}
           withPendingLoans={item.has_pending_loans}
           showPreviousLoan
+          checkIn
```
Comment (kprzerwa) [inline src/lib/pages/backoffice/Actions/CheckIn/CheckedInItems/CheckedInItems.js:22]: you didn't rename this one

File: src/lib/pages/frontsite/Documents/DocumentDetails/DocumentPanel/DocumentPanel.js [hunk 18]
```diff
@@ -65,13 +65,17 @@ class DocumentPanel extends Component {
...
-                      {doc.metadata.imprint?.publisher
-                        &&
-                      <><div
-                        className={`default-margin-bottom`}>
-                        Published by <b>{doc.metadata.imprint?.publisher}</b>
-                        {doc.metadata.publication_year && (<b>{doc.metadata.publication_year}</b>)}
-                      </div></>}
+                      {doc.metadata.imprint?.publisher && (
+                        <>
+                          <div className={`default-margin-bottom`}>
```
Comment (kprzerwa) [inline src/lib/pages/frontsite/Documents/DocumentDetails/DocumentPanel/DocumentPanel.js:70]:
```suggestion
                          <div className="default-margin-bottom">
```

File: src/lib/pages/frontsite/Documents/DocumentDetails/DocumentPanel/DocumentPanelMobile.js [hunk 19]
```diff
@@ -50,13 +50,16 @@ class DocumentPanelMobile extends Component {
...
-                  {doc.metadata.imprint?.publisher
-                    &&
-                  <><div
-                    className={`default-margin-bottom`}>
-                    Published by <b>{doc.metadata.imprint?.publisher}</b>
-                    {doc.metadata.publication_year && (<b>{doc.metadata.publication_year}</b>)}
-                  </div></>}
+                  {doc.metadata.imprint?.publisher && (
+                    <>
+                      <div className={`default-margin-bottom`}>
```
Comment (kprzerwa) [inline src/lib/pages/frontsite/Documents/DocumentDetails/DocumentPanel/DocumentPanelMobile.js:55]:
```suggestion
                      <div className="default-margin-bottom">
```
